src/app/ReadClient.cpp

```diff
@@ -143,7 +143,7 @@ CHIP_ERROR ReadClient::ScheduleResubscription(uint32_t aTimeTillNextResubscripti
         mReadPrepareParams.mSessionHolder.Grab(aNewSessionHandle.Value());
     }
 
-    mDoCaseOnNextResub = aReestablishCASE;
+    mForceCaseOnNextResub = aReestablishCASE;
```

Comment:
With the `DefaultResubscribePolicy`, the only way we can end up with `aReestablishCASE == false` is if `aTerminationCause == CHIP_ERROR_TIMEOUT` is not true.

Are we fixing a path where `aTerminationCause` is not `CHIP_ERROR_TIMEOUT`?

Reply:
I think I am looking for clarification specifically on this:

> We could end up in a situation where we had a defunct session but did not have the "try to re-establish CASE" flag set. In that case we would keep trying resubscribe attempts, which would keep failing because we could not actually create an exchange on our session, and we would never recover from that.

Examining the code, it does seem fragile to only attempt to reestablish CASE for `CHIP_ERROR_TIMEOUT`.  But I am trying to understand what paths enter this code with other causes.

Reply:
Yes.  We had a situation where we sent the Sigma1 and the server responded with an error status, and after that things were stuck.

Reply:
OK, makes sense.